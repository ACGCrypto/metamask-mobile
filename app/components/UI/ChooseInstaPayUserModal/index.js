import React from 'react';
import PropTypes from 'prop-types';
import { StyleSheet, View, ActivityIndicator } from 'react-native';
import Modal from 'react-native-modal';
import { colors, baseStyles } from '../../../styles/common';
import StyledButton from '../StyledButton';

const styles = StyleSheet.create({
	modal: {
		margin: 0,
		width: '100%'
	},
	modalView: {
		flexDirection: 'column',
		justifyContent: 'center',
		alignItems: 'center'
	},
	modalText: {
		alignSelf: 'center',
		width: '90%',
		height: 300,
		backgroundColor: colors.white,
		borderRadius: 10
	},
	actionContainer: {
		borderTopColor: colors.grey200,
		borderTopWidth: 1,
		flex: 0,
		flexDirection: 'row',
		padding: 16
	},
	button: {
		flex: 1
	},
	confirm: {
		marginLeft: 8
	}
});

/**
 * View that renders an action modal
 */
export default function ChooseInstaPayUserModal({ children, onConfirmPress, modalVisible, confirmText, loading }) {
	return (
		<Modal isVisible={modalVisible} style={styles.modal}>
			<View style={styles.modalView}>
				<View style={styles.modalText}>
					<View style={baseStyles.flexGrow}>{children}</View>
					<View style={styles.actionContainer}>
						<StyledButton
							type={'confirm'}
							onPress={onConfirmPress}
							containerStyle={[styles.button, styles.confirm]}
						>
							{loading ? <ActivityIndicator size="small" color="white" /> : confirmText.toUpperCase()}
						</StyledButton>
					</View>
				</View>
			</View>
		</Modal>
	);
}

ChooseInstaPayUserModal.defaultProps = {
	confirmText: 'CONFIRM'
};

ChooseInstaPayUserModal.propTypes = {
	/**
	 * Content to display above the action buttons
	 */
	children: PropTypes.node,
	/**
	 * Text to show in the confirm button
	 */
	confirmText: PropTypes.string,
	/**
	 * Called when the confirm button is clicked
	 */
	onConfirmPress: PropTypes.func,

	/**
	 * Whether modal is shown
	 */
	modalVisible: PropTypes.bool,
	/**
	 * Action triggered state
	 */
	loading: PropTypes.bool
};
