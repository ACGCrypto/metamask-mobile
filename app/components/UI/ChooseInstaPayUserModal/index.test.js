import React from 'react';
import { Text } from 'react-native';
import { shallow } from 'enzyme';
import ChooseInstaPayUserModal from './';

describe('ChooseInstaPayUserModal', () => {
	it('should render correctly', () => {
		const wrapper = shallow(
			<ChooseInstaPayUserModal
				// eslint-disable-next-line react/jsx-no-bind
				onConfirmPress={() => true}
				modalVisible
				confirmText={'Confirm'}
				loading={false}
			>
				<Text>Please wait</Text>
			</ChooseInstaPayUserModal>
		);
		expect(wrapper).toMatchSnapshot();
	});
});
