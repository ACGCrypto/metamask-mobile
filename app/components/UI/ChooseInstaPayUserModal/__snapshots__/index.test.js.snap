// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChooseInstaPayUserModal should render correctly 1`] = `
<ReactNativeModal
  animationIn="slideInUp"
  animationInTiming={300}
  animationOut="slideOutDown"
  animationOutTiming={300}
  avoidKeyboard={false}
  backdropColor="black"
  backdropOpacity={0.7}
  backdropTransitionInTiming={300}
  backdropTransitionOutTiming={300}
  deviceHeight={null}
  deviceWidth={null}
  hasBackdrop={true}
  hideModalContentWhileAnimating={false}
  isVisible={true}
  onBackButtonPress={[Function]}
  onBackdropPress={[Function]}
  onModalHide={[Function]}
  onModalShow={[Function]}
  onModalWillHide={[Function]}
  onModalWillShow={[Function]}
  scrollOffset={0}
  scrollOffsetMax={0}
  scrollTo={null}
  style={
    Object {
      "margin": 0,
      "width": "100%",
    }
  }
  supportedOrientations={
    Array [
      "portrait",
      "landscape",
    ]
  }
  swipeThreshold={100}
  useNativeDriver={false}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "column",
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "#FFFFFF",
          "borderRadius": 10,
          "height": 300,
          "width": "90%",
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <Text>
          Please wait
        </Text>
      </View>
      <View
        style={
          Object {
            "borderTopColor": "#bbc0c5",
            "borderTopWidth": 1,
            "flex": 0,
            "flexDirection": "row",
            "padding": 16,
          }
        }
      >
        <StyledButton
          containerStyle={
            Array [
              Object {
                "flex": 1,
              },
              Object {
                "marginLeft": 8,
              },
            ]
          }
          disabledContainerStyle={
            Object {
              "opacity": 0.6,
            }
          }
          onPress={[Function]}
          styleDisabled={
            Object {
              "opacity": 0.6,
            }
          }
          type="confirm"
        >
          CONFIRM
        </StyledButton>
      </View>
    </View>
  </View>
</ReactNativeModal>
`;
